mlir/lib/ExecutionEngine/ROCm/rocm-runtime-wrappers.cpp

```diff
@@ -499,7 +499,7 @@ mcpuMemset5DFloatRandInt(float *allocated, float *aligned, int64_t offset,
                          int64_t size0, int64_t size1, int64_t size2,
                          int64_t size3, int64_t size4, int64_t stride0,
                          int64_t stride1, int64_t stride2, int64_t stride3,
-                         int64_t stride4, short min, short max, int64_t seed) {
+                         int64_t stride4, short min, short max, int seed) {
```

Comment:
Maybe go with `int32_t` to be conservative?

Reply:
https://en.cppreference.com/w/cpp/numeric/random/srand
Looks srand expects 'unsigned' as seed type, I guess better not specify the size then.

Reply:
@jungpark-mlir Problem is that the populated test IR will generate a i32. I'm indicating to match based on that. But your observation is also valid. Guess this chunk of code was just poorly written. The populated code as well as the function signature should both be unsigned.

At this case, I'm less worried about the stl function's signature because it will do a implicit type conversion. However the runtime function matters more because we have no such cast in the populated IR.

Reply:
I see, thanks!

Reply:
I’ll amend the PR so:

1) rand in mlir-miopen-driver is always a positive integer or 0.
2) in case rand is 0, use srand(time()), otherwise use the specified number as the random seed.
3) change relevant documentation amd test cases.
4) use uint32_t here.

Reply:
@whchung Thanks, that sounds much better. We wouldn't be accidently pass a negative to srand after your change.